src/components/Cluster/NewCluster/AvailabilityZonesSelector.js

```diff
@@ -0,0 +1,83 @@
+import styled from '@emotion/styled';
+import PropTypes from 'prop-types';
+import React from 'react';
+import cmp from 'semver-compare';
+import NumberPicker from 'UI/NumberPicker';
+
+const FlexWrapperAZDiv = styled.div`
+  display: flex;
+  justify-content: flex-start;
+  align-items: center;
+
+  p {
+    margin-right: 18px;
+    transform: translateY(-4px);
+  }
+`;
+
+const AvailabilityZonesSelector = ({
+  minValue,
+  maxValue,
+  value,
+  readOnly,
+  onChange,
+  requiredReleaseVersion,
+  currentReleaseVersion,
+}) => {
+  const isAvailable = cmp(currentReleaseVersion, requiredReleaseVersion) >= 0;
```

Comment:
Is there a way we can change this to being computed on the cluster object (computeCapabilities) ?

Something like  `cluster.hasAvailabilityZones` ?

The goal is to move this kind of release and provider checking logic to one place.

It might mean that for this component we cant show the exact release version that would be required to support this feature, but I think that's ok.

So code like this: https://github.com/giantswarm/happa/pull/899/files#diff-d6aab637bd5ecfe4e323d3e499b95dddR77

Would join the computeCapabilities block.

Reply:
I'll add that now 👍

Reply:
That would work if we were fetching any clusters from the server. This check is made during cluster creation, where no clusters are fetched, so the only way we could use `computeCapabilities` is if we would use it separately, as an utility function.

Reply:
Oh yeah! True. Ok lets leave it as is for now. Maybe we refactor computeCapabilities to take a release version and a provider, instead of a cluster and a provider. Because I think those are the things we really check on. That said, lets see how computeCapabilities changes after @ograu is done refactoring and then revisit this idea.

There are a few other places I think where we still do version checks in 'view' components, and I'd like to move that out. Maybe we can tackle it in a more coordinated effort.

But I'm fine for leaving it in this PR for now.

Reply:
> That would work if we were fetching any clusters from the server. This check is made during cluster creation, where no clusters are fetched, so the only way we could use `computeCapabilities` is if we would use it separately, as an utility function.

We are actually fetching cluster details right after creation... check `batchedClusterCreate()`

Reply:
Yeah thats for getting to the cluster detail screen.

But this component is also used on the create cluster screen, where we don't have a cluster object yet. 


So computeCapabiltiies isn't the right function for this. I mean you could try to pass in a faked cluster object with just a release version set on it....

But what I think we should really do, is make a pure function, that doesn't modify the 'cluster' object that we pass in, but just returns a capabilities object. And then the implementing code can decide whether or not it wants to assign that to anything.

If what I'm saying is not making any sense, don't worry. I think we need to let the dust settle again a bit.

Lets talk in sig front end sync tomorrow? Or if you feel it is conflcting with the direction you're going in now, lets get on a call today.

Reply:
Nono, I think this is ok, I am more concerned about the flow, I mean __where__ this code is executed, and I think that this definitely should be a pure function probably outside actions files.